lib/collection/variable.js

```diff
@@ -208,6 +222,7 @@ _.assign(Variable.prototype, /** @lends Variable.prototype */ {
         _.has(options, 'system') && (this.system = options.system);
         _.has(options, 'disabled') && (this.disabled = options.disabled);
         _.has(options, 'description') && (this.describe(options.description));
+        _.has(options, 'lazy') && (this.lazy = options.lazy);
```

Comment:
@appurva21 Thoughts on this approach?
The advantage is that it makes it clear which variables are to be resolved lazily, and which are to be replaced immediately. The disadvantage is that `lazy` has no meaning for `replaceSubstitutions` ( non lazy ), and makes the `replaceSubstitutionsLazy` interface more complicated.

Some other approaches I thought about
- A) lazily replace all values ( when invoked through `replaceSubstitutionsLazy` ) but this increases the amount of memory needed, and slightly slows down the function.
- B) Only lazily replace `Promise` values. This is a good solution but there is no reliable method of identifying if something is a promise or not. ( Duck typing is not accurate )
- C) Only lazily replace function values. Could be a good tradeoff but has the same problems as (A), also makes the code slightly less intuitive.